### ISSUE TITLE
3.0.0 release breaks imglib2-roi

### ISSUE THREAD

**Issue by ctrueden** [imglib/imglib2-realtransform#28]
When upgrading imglib2-realtransform to 3.0.0, the imglib2-roi project now has a failing test:
```
java.lang.AssertionError: Dimensions do not match.
	at net.imglib2.realtransform.PositionFieldTransform.<init>(PositionFieldTransform.java:93)
	at net.imglib2.realtransform.DeformationFieldTransform.<init>(DeformationFieldTransform.java:64)
	at net.imglib2.roi.geom.OperatorsTest.testNonInvertibleTransform(OperatorsTest.java:808)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.eclipse.jdt.internal.junit4.runner.JUnit4TestReference.run(JUnit4TestReference.java:89)
	at org.eclipse.jdt.internal.junit.runner.TestExecution.run(TestExecution.java:41)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:542)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.runTests(RemoteTestRunner.java:770)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.run(RemoteTestRunner.java:464)
	at org.eclipse.jdt.internal.junit.runner.RemoteTestRunner.main(RemoteTestRunner.java:210)
```

Is it possible to restore backwards compatibility? Or is this something that needs to be adjusted in consumer code? If so, how to adjust?

**Comment by bogovicj** [comment]
On it

**Comment by bogovicj** [comment]
@ctrueden ,

@axtimwalde and I decided not to try to maintain backward compatibility (constructors would necessarily clash), and just break things sooner before too many things depend on it.

The PR for imglib2-roi above shows how to adjust the consumer code in response this breaking change.

**Comment by ctrueden** [comment]
Addressed by imglib/imglib2-roi#53—thanks @bogovicj!
